src/sparseml/transformers/utils/sparse_model.py

```diff
@@ -260,23 +277,32 @@ def text_generation_from_pretrained(
                 model_name_or_path
             )
 
-        kwargs["config"].is_decoder = True
-        kwargs["config"].use_past = False
-
         model = AutoModelForCausalLM.from_pretrained(
-            model_name_or_path,
-            **kwargs,
+            pretrained_model_name_or_path=model_name_or_path,
+            torch_dtype=torch_dtype,
         )
+        config = kwargs.get("config")
+        if config:
+            sequence_length = resolve_sequence_length(config)
+            model.seqlen = sequence_length
+        # TODO: Is adding `model.train()` here necessary? The instantiated model
+        # should automatically be in train mode
+        if recipe:
+            apply_structure_to_transformers(
```

Comment:
also loads the weights

Reply:
assumption is that 'recipy.yaml' exists side-by-side with the model.

Reply:
throw a warning if this is not the case.

Reply:
if you are training and fine tuning -> train mode,
if you are doing one shot or export -> eval mode,